crates/storage/src/tests.rs

```diff
@@ -15,6 +15,35 @@ async fn delete_nonexistent_key() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+/// In rare cases, the database lock has not been (yet) released by the time
+/// the next Storage::load() call is made. This is fixed by arranging the fields
+/// in Storage to be dropped in the right order. If this test fails, make sure
+/// to check that the dispatcher `Sender` is dropped first.
+async fn db_lock_is_released() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::init();
+    let tmpdir = tempfile::tempdir()?;
+
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
+    let storage = Storage::load(tmpdir.path().to_owned()).await?;
+    storage.release().await;
```

Comment:
This is the amount of consecutive db open/close that it took to consistently reproduce the `LOCK` contention on RocksDB across machines and architectures. Now the number doesn't matter because we wait for resources to be released before we try to re-open the database.